tests/VirtoCommerce.Platform.Caching.Tests/CacheDisablerTests.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Threading.Tasks;
+using VirtoCommerce.Platform.Core.Caching;
+using Xunit;
+
+namespace VirtoCommerce.Platform.Caching.Tests
+{
+    public class CacheDisablerTests
+    {
+        [Fact]
+        public void DisablesInCurrentThread()
+        {
+            Assert.False(CacheDisabler.CacheDisabled, "CacheDisabler shouldn't be active in this thread before test.");
+
+            using (CacheDisabler.DisableCache())
+            {
+                Assert.True(CacheDisabler.CacheDisabled, "CacheDisabler should be active in this thread.");
+            }
+
+            Assert.False(CacheDisabler.CacheDisabled, "CacheDisabler shouldn't be active in this thread after test.");
+        }
+
+        [Fact]
+        public void InheritedSuppressInAnotherThread()
+        {
+            Assert.False(CacheDisabler.CacheDisabled,
+                "CacheDisabler shouldn't be active in this thread before test.");
+
+            using (CacheDisabler.DisableCache())
+            {
+                Assert.True(CacheDisabler.CacheDisabled, "CacheDisabler should be active in this thread.");
+                Task.Run(() =>
```

Comment:
May you rewrite tests using real `async/await` and exiting/entering async code. Now it's really hard to understand what some tests doing.